libminifi/include/utils/HTTPUtils.h

```diff
@@ -31,14 +31,14 @@ so we convert localhost to our local hostname.
 inline bool parse_http_components(const std::string &url, std::string &port, std::string &scheme, std::string &path) {
 #ifdef WIN32
   auto hostname = (url.find(org::apache::nifi::minifi::io::Socket::getMyHostName()) != std::string::npos ? org::apache::nifi::minifi::io::Socket::getMyHostName() : "localhost");
-  std::string regex_str = "^(http|https)://(" + hostname + ":)([0-9]+)?(/.*)$";
+  std::string regex_str = "(http|https)://(" + hostname + ":)([0-9]+)?(/.*)$";
 #else
-  std::string regex_str = "^(http|https)://(localhost:)([0-9]+)?(/.*)$";
+  std::string regex_str = "(http|https)://(localhost:)([0-9]+)?(/.*)$";
```

Comment:
Please remove the trailing `$` sign / end of line match as well. It should no longer be necessary, just like `^`.

Reply:
Good catch, removed in 4d1784c2a28e7e9982f71ba3c77d0fac4f88c0c4